utils.py

```diff
@@ -74,12 +75,20 @@ def torchtext_iterators(
 
     # Load pretrained embeddings.
     if fasttext_embeds_path is not None:
+        embedding_vectors = torchtext.vocab.Vectors(name=fasttext_embeds_path)
         trg_field.vocab.load_vectors(
-            vectors=torchtext.vocab.Vectors(name=fasttext_embeds_path)
+            vectors=embedding_vectors
         )
         # TODO Remove words without embedding from the output vocabulary
         trg_field.vocab.vectors[trg_field.vocab.stoi[BOS_TOKEN]] = 1/16
         trg_field.vocab.vectors[(trg_field.vocab.vectors == 0).all(-1)] = trg_field.vocab.vectors.mean(0)
+        if is_vivo:
+            full_vocab = torchtext.data.Field()
+            full_vocab.build_vocab(train.trg, min_freq=min_freq)
+            excluded_words = [word for word in full_vocab.vocab.stoi if word not in trg_field.vocab.stoi]
+            excluded_vectors = torch.stack([embedding_vectors[excluded_word] for excluded_word in excluded_words if excluded_word in embedding_vectors.stoi])
```

Comment:
If this list is empty, PyTorch complains and throws an error. This happens when I set `--max_size` to `40` or lower. It looks like `full_vocab` and `trg_field.vocab` are constructed in almost the same way which would mean that their set difference is `{}`.

Reply:
They are not though, full_vocab does not have a max size

Reply:
For the error I could add something, but I did not have that bug occur. How did you produce it?

Reply:
```
python main.py \
    --device cuda \
    --checkpoint_path checkpoint_path \
    --fasttext_embeds_path corpus.fasttext.txt \
    --write_to_file \
    --eval_epochs 1 \
    --train_epochs 1 \
    --batch_size 64 \
    --max_len 40 \
    --loss_function vmf \
    --input_feed \
    --tie_embed \
    --enc_embed_size 300 \
    --mode train
```

Reply:
So maybe the error only occurs when your vocabulary is less than the max size.

Reply:
The vocab size for me is 112,000, which is twice the size of the max. I'll just check for an empty list in that case and return a random vector

Reply:
Even if you run with a `--max_len` of 40 or even 15?

Reply:
Ok I just pushed a verision that should fix your issue

Reply:
Yeah it works with short lengths now

Reply:
max len of 15,20,40 at the least